src/gpu/d3d12/SDL_gpu_d3d12.c

```diff
@@ -1446,11 +1460,15 @@ static void D3D12_INTERNAL_TextureSubresourceBarrier(
         textureSubresource->parent->container->header.info.usageFlags & SDL_GPU_TEXTUREUSAGE_COMPUTE_STORAGE_WRITE_BIT);
 }
 
-static D3D12_RESOURCE_STATES D3D12_INTERNAL_DefaultTextureResourceState(SDL_GpuTextureUsageFlags usageFlags)
+static D3D12_RESOURCE_STATES D3D12_INTERNAL_DefaultTextureResourceState(
+    SDL_GpuTextureUsageFlags usageFlags,
+    SDL_bool isSwapchainTexture)
 {
     /* NOTE: order matters here! */
 
-    if (usageFlags & SDL_GPU_TEXTUREUSAGE_SAMPLER_BIT) {
+    if (isSwapchainTexture) {
+        return D3D12_RESOURCE_STATE_PRESENT;
```

Comment:
If we're gonna do this, we should probably change the resource barrier in AcquireSwapchainTexture to use TextureSubresourceTransitionFromDefaultUsage.